sycl/include/CL/sycl/exception.hpp

```diff
@@ -12,40 +12,48 @@
 
 #include <CL/sycl/detail/common.hpp>
 #include <CL/sycl/stl.hpp>
-#include <exception>
+
+#include <cstddef>
 
 namespace cl {
 namespace sycl {
 
+// Forward declaration
 class context;
 
 struct exception {
+
+  using ContextPtr = shared_ptr_class<context>;
+
   exception() = default;
 
-  const char *what() const noexcept { return msg.c_str(); }
+  exception(const char *Msg, const cl_int CLErr = CL_SUCCESS,
+            ContextPtr Context = nullptr)
+      : Msg(string_class(Msg) + " " +
+            ((CLErr == CL_SUCCESS)
```

Comment:
I don't understand why do you need special handling for CL_SUCCESS, as I can see, it can be correctly handled by `OCL_CODE_TO_STR`

Reply:
Done.